perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2331,6 +2353,19 @@ public Member sponsorMember(PerunSession session, Member sponsoredMember, User s
 		return getMembersManagerImpl().getSponsoredMembers(sess, vo);
 	}
 
+	@Override
+	public Map<Member, List<RichUser>> getSponsoredMembersAndTheirSponsors(PerunSession sess, Vo vo) {
+		Map<Member, List<RichUser>> sponsoredMembersAndTheirSponsors = new HashMap<>();
+		List<Member> sponsoredMembers = getSponsoredMembers(sess, vo);
+		List<User> sponsors;
+		for (Member sponsoredMember : sponsoredMembers) {
+			sponsors = getPerunBl().getUsersManagerBl().getSponsors(sess, sponsoredMember);
+			sponsoredMembersAndTheirSponsors.put(sponsoredMember, convertUsersToRichUsers(sess, sponsors));
+			sponsors.clear();
+		}
```

Comment:
How about simplifying it
```java
for (Member sponsoredMember : sponsoredMembers) {
	sponsoredMembersAndTheirSponsors.put(sponsoredMember, getPerunBl().getUsersManagerBl().convertUsersToRichUsers(sess, getPerunBl().getUsersManagerBl().getSponsors(sess, sponsoredMember)));
}
```

Reply:
We decided to implemet it only in entry layer, so I changed it little bit more. I also simplified it by using stream.